src/main/java/teammates/ui/webapi/endpoints/WebPageEndpoints.java

```diff
@@ -0,0 +1,51 @@
+package teammates.ui.webapi.endpoints;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import teammates.common.util.Const.WebPageURIs;
+
+/**
+ * API endpoints for web pages.
+ */
+public enum WebPageEndpoints {
```

Comment:
Question: Is this a good design? (i.e. let the backend to decide what the frontend URL should be). Looks like it should be other way around. the control of frontend URL should be in the frontend rather than the backend.

Reply:
Isn't this the current design implementation as well? The URLs are defined in the `Const.java` file, and are matched by the frontend string literals.

I'm not sure if there are any gradle plugins that can convert TypeScript enums to Java enums, but I'll look into it.

Reply:
> Isn't this the current design implementation as well?

You are syncing this file `WebPageEndpoints.java` to the frontend right? In what situation will the frontend use this file? You might argue that this file can be used for any redirections in the frontend. But my question is, if one days the redirection changes, you should also modify the file in the backend which is a bit weired.

Reply:
I see what you mean. My intention was to sync both endpoints on the frontend and backend, so I thought it would be best to use a single source of truth in `Const.java`. 

If that's the case, should I just remove `WebPageEndpoints.java` entirely?